collections/sum_of.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+import { Selector } from "./types.ts";
+
+/**
+ * Applies the given selector to all elements in the given collection and calculates the sum of the results
+ *
+ * Example:
+ *
+ * ```typescript
+ * import { sumOf } from "./sum_of.ts"
+ *
+ * const people = [
+ *     { name: 'Anna', age: 34 },
+ *     { name: 'Kim', age: 42 },
+ *     { name: 'John', age: 23 },
+ * ]
+ * const totalAge = sumOf(people, i => i.age)
+ *
+ * console.assert(totalAge === 99)
+ * ```
+ */
+export function sumOf<T>(
+  array: Array<T>,
+  selector: Selector<T, number>,
```

Comment:
This was actually just in the proposal for internal clarity - I would suggest "flattening" the argument types, as the language server will otherwise just show `Selector`, which does not really help when you want to use the function quickly. I would suggest using a flat type in the args:

```suggestion
/**
 * Applies the given selector to all elements in the given collection and calculates the sum of the results
 *
 * Example:
 *
 * ```typescript
 * import { sumOf } from "./sum_of.ts"
 *
 * const people = [
 *     { name: 'Anna', age: 34 },
 *     { name: 'Kim', age: 42 },
 *     { name: 'John', age: 23 },
 * ]
 * const totalAge = sumOf(people, i => i.age)
 *
 * console.assert(totalAge === 99)
 * ```
 */
export function sumOf<T>(
  array: Array<T>,
  selector: (el: T) => number,
```